### PR TITLE
Abnormal security

### PR DESCRIPTION
**Pull request by dschuan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Update Abnormal Security Content Pack to V1.1.0

##### Updates
- Abuse Mailbox Information commands added
- Case Analysis Command added
- Genome/Employee Information command added
- Commands to retrieve threat/abuse mailbox logs in csv format added
- Enable optional subtenancy input when accessing threats, cases and abuse mailbox endpoints

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/README.md [hunk 1]
```diff
@@ -365,16 +460,19 @@ Get details of an Abnormal case
 
 
 #### Command Example
-```!abnormal-security-get-abnormal-case case_id=1234```
+```!abnormal-security-get-abnormal-case case_id=12805```
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/README.md:463]:
```suggestion
```!abnormal-security-get-abnormal-case case_id=1234```
```

**Review by Shellyber — CHANGES_REQUESTED** [review]
Great job on this contribution!
Please see my comments. 

Also let's schedule a demo for the new commands.

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/test_data/test_download_data_from_threat_log_in_csv_format.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Date,From Display Name,From Email,Subject,Recipient Email,Original Folder,Job Title of Sender,Job Title of Recipient,Abnormal's best guess of attack type,Abnormal's best guess of attack strategy,Abnormal's best guess of impersonated entity,Abnormal's best guess of attack vector,Remediation Status
+2021-07-20 20:14:32+00:00,'Kevin Lau',abc@def.com,[LIKELY PHISHING] does this go to junk,,,,,Email Account Takeover,Internal Compromised Email Account,None / Others,Text,Auto-Remediated to Junk
```
[the remark this answers is not in the document]
Comment (dschuan) [reply Packs/AbnormalSecurity/Integrations/AbnormalSecurity/test_data/test_download_data_from_threat_log_in_csv_format.csv:2]: it is provided in our mock data so it is cleared for use as an example :)

**Comment by dschuan** [comment]
@Shellyber thanks for the suggestions. Will make the changes and arrange a meeting with you separately.

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 3]
```diff
@@ -210,11 +316,48 @@ def get_details_of_an_abnormal_case_command(client, args):
...
-def get_the_latest_threat_intel_feed_command(client, args=None):
+def get_details_of_an_abuse_mailbox_campaign_command(client, args):
+    campaign_id = str(args.get('campaign_id', ''))
+    subtenant = args.get('subtenant', None)
 
-    response = client.get_the_latest_threat_intel_feed_request()
+    response = client.get_details_of_an_abuse_mailbox_campaign_request(campaign_id, subtenant)
+    command_results = CommandResults(
+        outputs_prefix='AbnormalSecurity.AbuseCampaign',
+        outputs_key_field='',
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:326]:
```suggestion
        outputs_key_field='campaignId',
```

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 4]
```diff
@@ -139,13 +222,35 @@ def get_a_list_of_abnormal_cases_identified_by_abnormal_security_command(client,
...
+    page_number = args.get('page_number', None)
+    subtenant = args.get('subtenant', None)
+
+    response = client.get_a_list_of_campaigns_submitted_to_abuse_mailbox_request(filter_, page_size, page_number, subtenant)
+    markdown = '### List of Abuse Mailbox Campaigns\n'
+    markdown += tableToMarkdown('Campaign IDs', response.get('campaigns', []), headers=['campaignId'])
+
+    command_results = CommandResults(
+        readable_output=markdown,
+        outputs_prefix='AbnormalSecurity.AbuseCampaigns',
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:237]:
```suggestion
        outputs_prefix='AbnormalSecurity.AbuseCampaign',
```

File: Packs/AbnormalSecurity/ReleaseNotes/1_1_0.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+
+#### Integrations
+##### Abnormal Security
+  - Features Added
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/ReleaseNotes/1_1_0.md:4]: As discussed, Let's remove the commands examples and leave only the commands.
For example:
- Abuse Mailbox Information commands:
  - abnormal-security-list-abuse-mailbox-campaigns
  - abnormal-security-get-abuse-mailbox-campaign

**Review by Shellyber — CHANGES_REQUESTED** [review]
Great job!
After the demo, please add `removeNull` to the `tableToMarkdown` function. 
and see my comments.

I'm will wait for your update regarding the `filter` argument.

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/command_examples [hunk 6]
```diff
@@ -2,9 +2,16 @@
 !abnormal-security-check-threat-action-status threat_id=xwvutsrq-9pon-mlkj-i876-54321hgfedcba action_id=abcdefgh-1234-5678-ijkl-mnop9qrstuvwx
 !abnormal-security-list-abnormal-cases filter="gte 2020-12-01T01:01:01Z"
 !abnormal-security-list-threats filter="gte 2020-12-01T01:01:01Z"
+!abnormal-security-download-threat-log-csv filter="receivedTime gte 2020-12-01T01:01:01Z"
```
Comment (dschuan) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/command_examples:5]: @Shellyber I'd input the filter wrongly during the demo (omitted the `receivedTime`) field. So we are good to go on the filter

**Comment by dschuan** [comment]
@Shellyber all changes complete on my end :) please let me know if there are any further changes recommended, thanks!

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.yml [hunk 7]
```diff
@@ -276,6 +361,123 @@ script:
...
+      name: mock-data
+      required: false
+    - description: Subtenant of the user (if applicable)
+      isArray: false
+      name: subtenant
+      required: false
+    description: Get details of an Abuse Mailbox campaign
+    name: abnormal-security-get-abuse-mailbox-campaign
+    outputs:
+      - contextPath: AbnormalSecurity.AbuseCampaignDetails.campaignId
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.yml:380]: Seems that after the changes, you didn't update the yml outputs.
Please fix according to the new one.

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 8]
```diff
@@ -210,11 +322,34 @@ def get_details_of_an_abnormal_case_command(client, args):
...
+        raw_response=response
+    )
+
+    return command_results
+
+
+def get_employee_identity_analysis_genome_data_command(client, args):
+    email_address = str(args.get('email_address', ''))
+
+    response = client.get_employee_identity_analysis_genome_data_request(email_address)
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:343]: Let's add to the response also the email_address (which is unique)
And then update the `outputs_key_field` in the command results with it.

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 9]
```diff
@@ -223,6 +358,42 @@ def get_the_latest_threat_intel_feed_command(client, args=None):
     return command_results
 
 
+def get_employee_information_command(client, args):
+    email_address = str(args.get('email_address', ''))
+
+    response = client.get_employee_information_request(email_address)
+    command_results = CommandResults(
+        outputs_prefix='AbnormalSecurity.Employee',
+        outputs_key_field='',
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:367]:
```suggestion
        outputs_key_field='email',
```

**Review by Shellyber — COMMENTED** [review]
@dschuan Looks good!
See my last comments

**Comment by dschuan** [comment]
@Shellyber thanks for the catch :) The suggestions have been implemented

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 10]
```diff
@@ -210,11 +322,35 @@ def get_details_of_an_abnormal_case_command(client, args):
...
+    headers = ['description', 'key', 'name', 'values']
+
+    markdown = tableToMarkdown(
+        f"Analysis of {email_address}", response.get('histograms', []), headers=headers, removeNull=True)
+
+    response["email"] = email_address
+    command_results = CommandResults(
+        readable_output=markdown,
+        outputs_prefix='AbnormalSecurity.Employee',
         outputs_key_field='',
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:354]:
```suggestion
        outputs_key_field='email',
```

**Comment by Shellyber** [comment]
@dschuan Please fix the conflict. and then we can merge

**Review by Shellyber — APPROVED** [review]

**Pull request merged**